cli/installer.rs

```diff
@@ -349,6 +348,24 @@ mod tests {
       Some("main".to_string())
     );
     assert_eq!(infer_name_from_url(&Url::parse("file:///").unwrap()), None);
+    assert_eq!(
+      infer_name_from_url(
+        &Url::parse("https://example.com/abc@0.1.0").unwrap()
+      ),
+      Some("abc".to_string())
+    );
+    assert_eq!(
+      infer_name_from_url(
+        &Url::parse("https://example.com/abc@0.1.0/main.ts").unwrap()
+      ),
+      Some("abc".to_string())
```

Comment:
I didn't even realize we already supported this fancy inference:
```
    assert_eq!(
      infer_name_from_url(&Url::parse("file:///abc/main.ts").unwrap()),
      Some("abc".to_string())
    );
```
I suppose given that this patch is useful.

Reply:
I feel a little queazy about the fancy inference tho. It's one of these magical algorithms that is just another thing to remember.